pkg/utils/oci/helmregistry.go

```diff
@@ -89,9 +89,8 @@ func buildRef(oci *gardencorev1.OCIRepository) (name.Reference, error) {
 		name.StrictValidation,
 	}
 
-	// in the local setup, we need to replace the registry and configure that we don't want to use TLS
-	if strings.Contains(ref, localRegistry) {
-		ref = strings.Replace(ref, localRegistry, inKubernetesRegistry, 1)
+	// in the local setup we don't want to use TLS
+	if strings.Contains(ref, inKubernetesRegistry) {
```

Comment:
You have to adapt the unit test https://github.com/gardener/gardener/blob/78565916bef348d472d635e0e1b2449ed6b335d9/pkg/utils/oci/helmregistry_test.go#L126-L149